Source/WebCore/page/Quirks.cpp

```diff
@@ -1568,11 +1569,36 @@ bool Quirks::shouldExposeShowModalDialog() const
 {
     if (!needsQuirks())
         return false;
+
     if (!m_shouldExposeShowModalDialog) {
         auto domain = RegistrableDomain(m_document->url()).string();
         m_shouldExposeShowModalDialog = domain == "pandora.com"_s || domain == "marcus.com"_s;
     }
     return *m_shouldExposeShowModalDialog;
 }
 
+bool Quirks::shouldDisableLazyImageLoading() const
+{
+    if (!needsQuirks())
+        return false;
+
+    if (m_shouldDisableLazyImageLoading)
+        return m_shouldDisableLazyImageLoading.value();
+
+    m_shouldDisableLazyImageLoading = false;
+
+    // Quirk only applies for IKEA.com.
+    // Searching for <meta name="generator" content="Gatsby 4.24.1" />
+    // Images will have a grey box over them when loaded lazily in large percentage of page loads.
+    // This issue is not observed when lazy image loading is disabled, and has been fixed in future Gatsby versions.
+    auto* metaElement = m_document->getElementsByTagName("meta"_s).get().namedItem("generator"_s);
+    
+    if (RegistrableDomain(m_document->url()).string() == "ikea.com"_s 
+        && metaElement 
+        && metaElement->getAttribute("content"_s) == "Gatsby 4.24.1"_s)
```

Comment:
Per WebKit coding style, if a condition is on multiple lines, you need to use curly brackets.